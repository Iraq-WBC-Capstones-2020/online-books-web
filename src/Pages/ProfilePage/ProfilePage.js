import React from 'react';
import { MDBContainer, MDBRow, MDBCol } from 'mdbreact';
import { useTranslation } from 'react-i18next';
import Paragraph from '../../components/Paragraph/Paragraph';
import Pills from '../../components/Pills/Pills';
import ProfileInfo from '../../components/ProfilePage/ProfileInfo/ProfileInfo';
import PurchaseHistory from '../../components/ProfilePage/PurchaseHistory/PurchaseHistory';
import { useSelector } from 'react-redux';

function ProfilePage() {
  const { t } = useTranslation();
  const user = useSelector((state) => state.user);
  const profileInfo = {
    id: user.id,
    name: user.fullName,
    email: user.email,
    img: 'https://via.placeholder.com/300x300',
    gender: 'Female',
    joined: new Date('02/01/2010').getTime(),
    birthday: new Date('02/01/1998').getTime(),
    phoneNumber: '07154245455',
    address: {
      street: 'Middle Erbil',
      city: 'Erbil',
      country: 'Iraq',
    },
  };
  const userParagraph = `${t('profilePage.memberSince')} ${new Date(
    profileInfo.joined
  ).toDateString()}\n
  ${t('profilePage.livesIn')} ${profileInfo.address.city}${t(
    'profilePage.addressSpread'
  )}${profileInfo.address.country}\n
  ${profileInfo.email}`;
  const tabs = [
    {
      title: t('profilePage.about'),
      content: <ProfileInfo profileInfo={profileInfo} />,
    },
    {
      title: t('profilePage.purchaseHistory'),
      content: <PurchaseHistory profileInfo={profileInfo} />,
    },
  ];
  return (
    <>
      <div className="bg-gray-200">
        <MDBContainer>
          <MDBRow className="h-auto">
            <MDBCol lg="3" md="6" sm="12" size="12" className="py-4">
              <img
                src={profileInfo.img}
                alt={profileInfo.name}
                className="rounded-lg mx-auto h-auto max-w-full"
              />
            </MDBCol>
            <MDBCol
              lg="8"
              md="6"
              sm="12"
              size="12"
              className="py-4 text-center text-md-left"
            >
              <h4 className="text-3xl font-medium text-blue-500 pb-3">
                {profileInfo.name}
              </h4>
              {userParagraph.split('\n').map((paragraph, index) => (
                <Paragraph className="text-center text-md-left" key={index}>
                  {paragraph}
                </Paragraph>
              ))}
            </MDBCol>
          </MDBRow>
        </MDBContainer>
      </div>
      <Pills tabs={tabs} />
    </>
  );
}

export default ProfilePage;
