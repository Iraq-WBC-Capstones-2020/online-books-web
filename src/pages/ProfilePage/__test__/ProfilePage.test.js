import React from 'react';
import renderer from 'react-test-renderer';
import ProfilePage from '../ProfilePage';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
test('ProfilePage snapshot', () => {
  jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key) => key }),
  }));
  const history = createMemoryHistory();

  const tree = renderer
    .create(
      <Router history={history}>
        <ProfilePage />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
