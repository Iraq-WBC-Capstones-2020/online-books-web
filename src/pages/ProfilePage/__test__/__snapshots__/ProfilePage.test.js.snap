// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage snapshot 1`] = `
Array [
  <div
    className="bg-gray-200"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="row h-auto"
        data-test="row"
      >
        <div
          className="col-12 col-sm-12 col-md-6 col-lg-3 py-4"
          data-test="col"
        >
          <img
            alt="Sara Ahmed"
            className="rounded-lg mx-auto h-auto max-w-full"
            src="https://via.placeholder.com/300x300"
          />
        </div>
        <div
          className="col-12 col-sm-12 col-md-6 col-lg-8 py-4 text-center text-md-left"
          data-test="col"
        >
          <h4
            className="text-3xl font-medium text-blue-500 pb-3"
          >
            Sara Ahmed
          </h4>
          <p
            className="text-3xl font-medium border-b-4 inline border-blue-500 text-center text-md-left"
          />
          <p
            className="text-base leading-loose text-justify text-center text-md-left"
          >
            profilePage.memberSince Mon Feb 01 2010
          </p>
          <p
            className="text-3xl font-medium border-b-4 inline border-blue-500 text-center text-md-left"
          />
          <p
            className="text-base leading-loose text-justify text-center text-md-left"
          >
            
          </p>
          <p
            className="text-3xl font-medium border-b-4 inline border-blue-500 text-center text-md-left"
          />
          <p
            className="text-base leading-loose text-justify text-center text-md-left"
          >
              profilePage.livesIn ErbilprofilePage.addressSpreadIraq
          </p>
          <p
            className="text-3xl font-medium border-b-4 inline border-blue-500 text-center text-md-left"
          />
          <p
            className="text-base leading-loose text-justify text-center text-md-left"
          >
            
          </p>
          <p
            className="text-3xl font-medium border-b-4 inline border-blue-500 text-center text-md-left"
          />
          <p
            className="text-base leading-loose text-justify text-center text-md-left"
          >
              sara@eample.com
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="py-2 border-t-2 border-b-2"
    >
      <div
        className="container"
        data-test="container"
      >
        <ul
          className="nav nav-pills"
          data-test="nav"
        >
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="nav-link Ripple-parent nav-link Ripple-parent active"
              data-test="nav-link"
              href="/"
              onClick={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
            >
              profilePage.about
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </a>
          </li>
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="nav-link Ripple-parent nav-link Ripple-parent"
              data-test="nav-link"
              href="/"
              onClick={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
            >
              profilePage.purchaseHistory
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="bg-gray-200 py-4"
    >
      <div
        className="container"
        data-test="container"
      >
        <div
          className="tab-content"
          data-test="tabContent"
        >
          <div
            className="tab-pane active"
            data-test="tab-pane"
            role="tabpanel"
          >
            <button
              className="btn-transparent btn Ripple-parent shadow-none text-blue-500 p-0 m-0 float-right mr-2 capitalize"
              data-test="button"
              onClick={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              profilePage.edit
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
            <div
              className="clearfix"
            />
            <div
              className="row"
              data-test="row"
            >
              <div
                className="col-12 col-sm-12 col-md-6"
                data-test="col"
              >
                <h3
                  className="text-2xl leading-loose font-semibold"
                >
                  profilePage.basicInfo
                </h3>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.name
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      Sara Ahmed
                    </h3>
                  </div>
                </div>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.gender
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      Female
                    </h3>
                  </div>
                </div>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.birthday
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      Sun Feb 01 1998
                    </h3>
                  </div>
                </div>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.city
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      Erbil
                    </h3>
                  </div>
                </div>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.country
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      Iraq
                    </h3>
                  </div>
                </div>
              </div>
              <div
                className="col-12 col-sm-12 col-md-6"
                data-test="col"
              >
                <h3
                  className="text-2xl leading-loose font-semibold"
                >
                  profilePage.contactInfo
                </h3>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.address
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      Middle Erbil
                    </h3>
                  </div>
                </div>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.phoneNumber
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      07154245455
                    </h3>
                  </div>
                </div>
                <div
                  className="row"
                  data-test="row"
                >
                  <div
                    className="col-6 col-sm-6 col-md-4"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      profilePage.email
                      :
                    </h3>
                  </div>
                  <div
                    className="col-6 col-sm-6 col-md-8"
                    data-test="col"
                  >
                    <h3
                      className="text-base leading-loose"
                    >
                      sara@eample.com
                    </h3>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="tab-pane"
            data-test="tab-pane"
            role="tabpanel"
          >
            <div
              className="mdb-datatable dt-bootstrap4"
              data-test="datatable"
            >
              <div
                className="row"
              >
                <div
                  className="col-sm-12"
                  data-test="mdb-datatable-table"
                >
                  <div
                    className=""
                    data-test="table"
                    style={
                      Object {
                        "maxHeight": undefined,
                      }
                    }
                  >
                    <table
                      className="table table-borderless table-hover mdb-dataTable"
                      entries={5}
                    >
                      <thead
                        className="mdb-dataTable-head"
                        data-test="datatable-head"
                      >
                        <tr>
                          <th
                            aria-controls="DataTable"
                            aria-label="orderId"
                            className="sorting"
                            onClick={[Function]}
                          >
                            profilePage.orderId
                          </th>
                          <th
                            className="sorting"
                            onClick={[Function]}
                          >
                            profilePage.books
                          </th>
                          <th
                            className="sorting"
                            onClick={[Function]}
                          >
                            profilePage.status
                          </th>
                          <th
                            className="sorting"
                            onClick={[Function]}
                          >
                            profilePage.originalPrice
                          </th>
                          <th
                            className="sorting"
                            onClick={[Function]}
                          >
                            profilePage.totalPrice
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        data-test="table-body"
                      >
                        <tr>
                          <td>
                            Tiger Nixon
                          </td>
                          <td>
                            System Architect
                          </td>
                          <td>
                            Edinburgh
                          </td>
                          <td>
                            61
                          </td>
                          <td>
                            2011/04/25
                          </td>
                        </tr>
                        <tr>
                          <td>
                            Tiger Nixon
                          </td>
                          <td>
                            System Architect
                          </td>
                          <td>
                            Edinburgh
                          </td>
                          <td>
                            61
                          </td>
                          <td>
                            2011/04/25
                          </td>
                        </tr>
                        <tr>
                          <td>
                            Tiger Nixon
                          </td>
                          <td>
                            System Architect
                          </td>
                          <td>
                            Edinburgh
                          </td>
                          <td>
                            61
                          </td>
                          <td>
                            2011/04/25
                          </td>
                        </tr>
                        <tr>
                          <td>
                            Tiger Nixon
                          </td>
                          <td>
                            System Architect
                          </td>
                          <td>
                            Edinburgh
                          </td>
                          <td>
                            61
                          </td>
                          <td>
                            2011/04/25
                          </td>
                        </tr>
                        <tr>
                          <td>
                            Tiger Nixon
                          </td>
                          <td>
                            System Architect
                          </td>
                          <td>
                            Edinburgh
                          </td>
                          <td>
                            61
                          </td>
                          <td>
                            2011/04/25
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="d-flex w-100 justify-content-between"
                >
                  <div
                    className="row ml-3 justify-content-between"
                  >
                    <div
                      className="false"
                      data-test="datatable-search"
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                        }
                      }
                    >
                      <div
                        className="mdb-datatable-filter flex-row"
                        data-test="datatable-input"
                      >
                        <input
                          aria-label="Search"
                          className="form-control form-control-sm ml-0 my-1"
                          onChange={[Function]}
                          placeholder="Search"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="d-flex align-items-center justify-content-end"
                  >
                    <div
                      className="mdb-datatable-entries"
                      data-test="mdb-datatable-entries"
                    >
                      <div
                        className="mdb-datatable-length bs-select"
                        data-test="datatable-select"
                      >
                        <label>
                          Rows per page:
                          <select
                            className="custom-select custom-select-sm form-control form-control-sm"
                            onChange={[Function]}
                            style={
                              Object {
                                "marginLeft": ".5rem",
                              }
                            }
                            value={5}
                          >
                            <option
                              value={5}
                            >
                              5
                            </option>
                            <option
                              value={20}
                            >
                              20
                            </option>
                            <option
                              value={25}
                            >
                              25
                            </option>
                          </select>
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdb-datatable-info d-flex align-items-center"
                      data-test="datatable-info"
                    >
                      <div
                        aria-live="polite"
                        role="status"
                      >
                         1 - 5 of 8 
                      </div>
                    </div>
                    <div
                      data-test="datatable-pagination"
                    >
                      <div
                        className="mdb-dataTables_paginate"
                      >
                        <ul
                          className="pagination m-0"
                          data-test="pagination"
                        >
                          <li
                            className="disabled page-item"
                            data-test="page-item"
                          >
                            <a
                              aria-label="Prev"
                              className="page-link page-link"
                              data-test="page-link"
                              onClick={[Function]}
                            >
                              <span>
                                <i
                                  className="fas fa-chevron-left"
                                />
                              </span>
                            </a>
                          </li>
                          <li
                            className="page-item"
                            data-test="page-item"
                          >
                            <a
                              aria-label="Next"
                              className="page-link page-link"
                              data-test="page-link"
                              onClick={[Function]}
                            >
                              <span>
                                <i
                                  className="fas fa-chevron-right"
                                />
                              </span>
                            </a>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
